tests/test_datasets/test_hierarchical_dataset.py

```diff
@@ -298,3 +454,94 @@ def test_to_hierarchical_dataset_hierarchical_structure(
+
+
+@pytest.mark.parametrize(
+    "hierarchical_structure_name,source_df_name,target_level,target_df_name",
+    (
+        ("hierarchical_structure", "product_level_df", "market", "market_level_df"),
+        ("hierarchical_structure", "product_level_df", "total", "total_level_df"),
+        ("hierarchical_structure", "market_level_df", "total", "total_level_df"),
+        ("hierarchical_structure", "product_level_df_w_nans", "market", "market_level_df_w_nans"),
+        ("hierarchical_structure", "product_level_df_w_nans", "total", "total_level_df_w_nans"),
+        ("hierarchical_structure", "market_level_df_w_nans", "total", "total_level_df_w_nans"),
+        ("long_hierarchical_structure", "l4_level_df_long", "l3", "l3_level_df_long"),
+        ("long_hierarchical_structure", "l4_level_df_long", "l2", "l2_level_df_long"),
+        ("long_hierarchical_structure", "l4_level_df_long", "l1", "l1_level_df_long"),
+        ("long_hierarchical_structure", "l3_level_df_long", "l2", "l2_level_df_long"),
+        ("long_hierarchical_structure", "l3_level_df_long", "l1", "l1_level_df_long"),
+        ("long_hierarchical_structure", "l2_level_df_long", "l1", "l1_level_df_long"),
+        ("tailed_hierarchical_structure", "l4_level_df_tailed", "l3", "l3_level_df_tailed"),
+        ("tailed_hierarchical_structure", "l4_level_df_tailed", "l2", "l2_level_df_tailed"),
+        ("tailed_hierarchical_structure", "l4_level_df_tailed", "l1", "l1_level_df_tailed"),
+        ("tailed_hierarchical_structure", "l3_level_df_tailed", "l2", "l2_level_df_tailed"),
+        ("tailed_hierarchical_structure", "l3_level_df_tailed", "l1", "l1_level_df_tailed"),
+        ("tailed_hierarchical_structure", "l2_level_df_tailed", "l1", "l1_level_df_tailed"),
+    ),
+)
+def test_get_level_dataset(hierarchical_structure_name, source_df_name, target_level, target_df_name, request):
+    hierarchical_structure = request.getfixturevalue(hierarchical_structure_name)
+
+    source_df = request.getfixturevalue(source_df_name)
+    source_ts = TSDataset(df=source_df, freq="D", hierarchical_structure=hierarchical_structure)
+
+    target_df = request.getfixturevalue(target_df_name)
+    target_ts = TSDataset(df=target_df, freq="D", hierarchical_structure=hierarchical_structure)
+
+    estimated_target_ts = source_ts.get_level_dataset(target_level)
+
+    # check attributes
+    assert target_ts.freq == estimated_target_ts.freq
+    assert target_ts.hierarchical_structure == estimated_target_ts.hierarchical_structure
+    assert target_ts.current_df_level == estimated_target_ts.current_df_level
+
+    pd.testing.assert_frame_equal(target_ts.df, estimated_target_ts.df)
+
+
+@pytest.mark.parametrize(
+    "source_df_name,target_level,target_df_name",
+    (
+        ("product_level_df", "market", "market_level_df"),
+        ("product_level_df", "total", "total_level_df"),
+        ("market_level_df", "total", "total_level_df"),
+    ),
+)
+def test_get_level_dataset_with_exog(
```

Comment:
Here also add regressor column in df_exog and use parameter `known_future` to pass it. Check that this attribute is also initialized correctly.